From 990bbd4805e82b1111a18bf3a3645cf9880f283f Mon Sep 17 00:00:00 2001
Message-Id: <990bbd4805e82b1111a18bf3a3645cf9880f283f.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Sat, 5 Jul 2014 01:33:51 +0530
Subject: [PATCH 095/312] REVERTME [VPG]: drm/i915: Quirky MIPI pipe config
 tracking

This patch helps tracking pipe config. This patch is currently
a REVERTME. Appropriate fix will be uploaded.

Issue: GMIN-1469
Change-Id: I8a9215f05537876695626aa1593e479bb9c17660
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++--
 drivers/gpu/drm/i915/intel_dsi.c     |   12 ++++++++++++
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 0c2db34..1f30078 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -7103,9 +7103,10 @@ static bool i9xx_get_pipe_config(struct intel_crtc *crtc,
 
 	if (IS_CHERRYVIEW(dev))
 		chv_crtc_clock_get(crtc, pipe_config);
-	else if (IS_VALLEYVIEW(dev))
+	else if (IS_VALLEYVIEW(dev) &&
+		!intel_pipe_has_type(&crtc->base, INTEL_OUTPUT_DSI))
 		vlv_crtc_clock_get(crtc, pipe_config);
-	else
+	else if (!IS_VALLEYVIEW(dev))
 		i9xx_crtc_clock_get(crtc, pipe_config);
 
 	return true;
diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 7c07ee0..554e76f 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -350,9 +350,21 @@ static bool intel_dsi_get_hw_state(struct intel_encoder *encoder,
 static void intel_dsi_get_config(struct intel_encoder *encoder,
 				 struct intel_crtc_config *pipe_config)
 {
+	struct drm_i915_private *dev_priv = encoder->base.dev->dev_private;
+
 	DRM_DEBUG_KMS("\n");
 
 	/* XXX: read flags, set to adjusted_mode */
+	pipe_config->quirks = 1;
+
+	/* for MIPI we use DSI PLL, not DPLL */
+	memset(&pipe_config->dpll_hw_state, 0,
+		sizeof(pipe_config->dpll_hw_state));
+
+	/* for now statically initialize dot clock */
+	pipe_config->adjusted_mode.crtc_clock =
+		dev_priv->vbt.lfp_lvds_vbt_mode->clock;
+	pipe_config->port_clock = dev_priv->vbt.lfp_lvds_vbt_mode->clock;
 }
 
 static enum drm_mode_status
-- 
1.7.9.5

